pages/studentPortal/labs/multiplication-connect/Index.tsx

```diff
@@ -1,126 +1,26 @@
-  isPlayerOne
-    ? (player = SelectedBy.PlayerOne)
-    : (player = SelectedBy.PlayerTwo);
-  let str: string;
-  for (let i = 0; i < rows.length; i++) {
-    // rows.length == board height == 7
-    // rows[i].length == board width == 5
-    for (let index = 0; index < rows[i].length - 3; index++) {
-      // Horizontal check
-      rows[i][index].selectedBy == player &&
-      rows[i][index + 1].selectedBy == player &&
-      rows[i][index + 2].selectedBy == player &&
-      rows[i][index + 3].selectedBy == player
-        ? (str = `${player} (horizontal) Four in a row!`)
-        : "";
-    }
-    if (i < rows.length - 3) {
-      // Vertical check
-      for (let index = 0; index < rows[i].length; index++) {
-        rows[i][index].selectedBy == player &&
-        rows[i + 1][index].selectedBy == player &&
-        rows[i + 2][index].selectedBy == player &&
-        rows[i + 3][index].selectedBy == player
-          ? (str = `${player} (vertical) Four in a row!`)
-          : "";
-      }
-    }
-    if (i >= 3) {
-      // Ascending diagonal check
-      for (let index = 0; index < rows[i].length - 3; index++) {
-        rows[i][index].selectedBy == player &&
-        rows[i - 1][index + 1].selectedBy == player &&
-        rows[i - 2][index + 2].selectedBy == player &&
-        rows[i - 3][index + 3].selectedBy == player
-          ? (str = `${player} (ascending diagonal) Four in a row!`)
-          : "";
-      }
-      // Descending diagonal check
-      for (let index = 3; index < rows[i].length; index++) {
-        rows[i][index].selectedBy == player &&
-        rows[i - 1][index - 1].selectedBy == player &&
-        rows[i - 2][index - 2].selectedBy == player &&
-        rows[i - 3][index - 3].selectedBy == player
-          ? (str = `${player} (descending diagonal) Four in a row!`)
-          : "";
-      }
-    }
-  }
-  str
-    ? ""
-    : grid.some((i) => i.selectedBy === SelectedBy.Unselected)
-    ? (str = "No winner")
-    : (str = "Draw");
-  // console.log(str);
-  return str;
-};
-
-export const createGrid = () => {
-  let arr = [];
-  let newGrid = [];
-  for (let i = 4; i < 25; i++) i % 2 === 0 ? arr.push(i) : "";
-  for (let i = 0; i < 35; i++) {
-    let gridNumber = getRandomItemFromArray(arr);
-    newGrid.push({
-      id: i,
-      gridNumber: gridNumber,
-      selectedBy: SelectedBy.Unselected,
-    });
-  }
-  return newGrid;
-};
+import { calculateWinner } from "../../../api/labs/games/multiplication-connect/gameLogic";
 
 const Index: FC = () => {
-  const [grid, setGrid] = useState([]);
   const [newGame, setNewGame] = useState(0);
-  const [isPlayerOne, setIsPlayerOne] = useState(true);
+  const { grid, isPlayerOne } = useSelector(multiplicationConnectSelector);
+  const dispatch = useDispatch();
 
   useEffect(() => {
     console.log(`newGame: ${newGame}`);
-    setGrid(createGrid);
-    setIsPlayerOne(true);
+    dispatch(reloadGrid(grid));
+    !isPlayerOne ? dispatch(togglePlayer(isPlayerOne)) : "";
```

Comment:
nice! redux works here. amazing.

Reply:
Please let me know if you have suggestions!